lib/primitives/mtx.js

```diff
@@ -1705,33 +1773,62 @@ class CoinSelector {
           this.inputs.delete(key);
         }
       }
+    }
 
-      if (this.inputs.size > 0)
-        throw new Error('Could not resolve preferred inputs.');
+    if (this.inputs.size > 0)
+      throw new Error('Could not resolve existing inputs.');
 
-      for (const coin of coins) {
-        this.tx.addCoin(coin);
-        this.chosen.push(coin);
-      }
+    for (const coin of coins) {
+      view.addCoin(coin);
+      this.viewCoins.push(coin);
```

Comment:
Why is this ending different than `resolvePreferred()`? Do we already have these coins in `chosen`?

```js
    for (const coin of coins) {
      this.tx.addCoin(coin);
      this.chosen.push(coin);
    }
```

Reply:
this.tx.addCoin will also create `inputs` in the tx. We don't want to do that here, because those inputs already exist in the `this.tx`. We are just updating view (Comment above)